micrometer-shim/src/main/java/io/opentelemetry/example/micrometer/Controller.java

```diff
@@ -1,21 +1,17 @@
 package io.opentelemetry.example.micrometer;
 
-import io.micrometer.core.annotation.Timed;
-import java.util.Random;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 public class Controller {
 
+  @Autowired private Service service;
+
   @GetMapping("/ping")
   public String ping() throws InterruptedException {
-    doWork();
+    service.doWork();
     return "pong";
   }
-
-  @Timed("dowork.time")
-  private void doWork() throws InterruptedException {
-    Thread.sleep(new Random().nextInt(1000));
-  }
```

Comment:
just curious, does `@Timed` not work here? I thought aop would apply to `@RestController` classes, so just checking my understanding, thx

Reply:
If I remember correctly, spring does not add aspects to private methods by default, you have to use aspectjweaver for that

Reply:
> If I remember correctly, spring does not apply aspects to private methods by default

Correct. Spring automatically times methods annotated with annotations like `@GetMapping`. With the changes in this PR, it times methods annotated with `@Timed`, as long as they're not private methods called from within the same class.